src/Codeception/Module/WebDriver.php

```diff
@@ -338,6 +340,19 @@ public function _before(TestInterface $test)
         ]);
     }
 
+    public function _reconfigure($config)
+    {
+        parent::_reconfigure($config);
+        if (count($config) === 1 && array_key_exists('url', $config)) {
+            //don't restart browser if only url was changed
+            return;
+        }
+        $this->reconfigured = true;
+        $this->webDriver->quit();
```

Comment:
I found an edge case here.

If we have a chrome browser running and `restart: false`.
When we are changing browser to firefox and restarting browser in all next tests we will have firefox running.
While `_reconfigure` should rollback its changes in the end of a test.